RecoJets/JetProducers/src/PileupJetIdAlgo.cc

```diff
@@ -525,35 +634,101 @@ PileupJetIdentifier PileupJetIdAlgo::computeIdVariables(const reco::Jet* jet,
     }
 
     // trailing candidate
-    if (lTrail == nullptr || candPt < lTrail->pt()) {
+    if ((!useBugFix) && (lTrail == nullptr || candPt < lTrail->pt())) {
       lTrail = icand;
     }
+    else if ((useBugFix) && (lTrail == nullptr || candPt < (lTrail->pt())*TrailcandWeight)){
+      lTrail = icand;
+      if (applyConstituentWeight) {
+        TrailcandWeight = constituentWeights[jet->sourceCandidatePtr(i)];
+      }
+    }
+
+  // average for pull variavble
+    if (useBugFix) {
+      float weight2 = candPt * candPt;
+      sumW2 += weight2;
+      float deta = icand->eta() - jet->eta();
+      float dphi = reco::deltaPhi(*icand, *jet);
+      sum_deta += deta * weight2;
+      sum_dphi += dphi * weight2;
+      if (sumW2 > 0) {
+        ave_deta = sum_deta / sumW2;
+        ave_dphi = sum_dphi / sumW2;
+      }
```

Comment:
Do not recalculate the averages in every iteration inside the loop

Reply:
Will move it outside of the loop and fix also in the `master` branch.